src/vue/forms/IssuanceForm.vue

```diff
@@ -0,0 +1,272 @@
+        :disabled="formMixin.isDisabled"
+        @click.prevent="emitCancel"
+      >
+        {{ 'issuance.cancel-btn' | globalize }}
+      </button>
+    </div>
+  </form>
+</template>
+
+<script>
+import AssetLoaderMixin from '@/vue/mixins/asset-loader.mixin'
+import FormMixin from '@/vue/mixins/form.mixin'
+
+import { Bus } from '@/js/helpers/event-bus'
+import { ErrorHandler } from '@/js/helpers/error-handler'
+
+import { Sdk } from '@/sdk'
+import { base } from '@tokend/js-sdk'
+
+import { required, amountRange, emailOrAccountId } from '@validators'
+
+const EVENTS = {
+  cancel: 'cancel'
+}
+
+export default {
+  name: 'issuance-form',
+  mixins: [AssetLoaderMixin, FormMixin],
+  data: _ => ({
+    form: {
+      asset: '',
+      amount: 0,
+      receiver: '',
+      reference: ''
+    }
+  }),
+  validations () {
+    return {
+      form: {
+        asset: { required },
+        amount: {
+          required,
+          amountRange: amountRange(
+            0.000001,
+            this.availableAmount ? this.availableAmount.value : 0
+          )
+        },
+        receiver: { required, emailOrAccountId },
+        reference: { required }
+      }
+    }
+  },
+  computed: {
+    assetListValues () {
+      return this.ownedAssets
+        .map(asset => ({
+          value: asset.code,
+          label: `${asset.details.name} (${asset.code})`
+        }))
+    },
+    availableAmount () {
+      if (this.ownedAssets.length && this.form.asset) {
+        const asset = this.ownedAssets
+          .find(asset => asset.code === this.form.asset)
+        return {
+          value: asset.availableForIssuance,
+          currency: asset.code
+        }
+      }
+      return { value: 0 }
+    }
+  },
+  async created () {
+    await this.loadOwnedAssets()
+    if (this.ownedAssets.length > 0) {
+      this.form.asset = this.ownedAssets[0].code
+    }
+  },
+  methods: {
+    async submit () {
+      if (!this.isFormValid()) return
+      this.disableForm()
+      try {
+        const receiverBalance =
+          await this.getReceiverBalance(this.form.receiver, this.form.asset)
+
+        if (receiverBalance) {
+          const operation =
+            base.CreateIssuanceRequestBuilder.createIssuanceRequest({
+              asset: this.form.asset,
+              amount: this.form.amount.toString(),
+              receiver: receiverBalance.balanceId,
+              reference: this.form.reference,
+              externalDetails: {}
+            })
+          await Sdk.horizon.transactions.submitOperations(operation)
+          Bus.success('issuance.tokens-issued-msg')
+          this.emitCancel()
+        } else {
+          Bus.error('issuance.balance-required-err')
+        }
+      } catch (e) {
+        console.error(e)
+        ErrorHandler.process(e)
+      }
+      this.enableForm()
+    },
+    async getAccountInfoByEmail (email) {
+      try {
+        const response = await Sdk.api.users.getPage({ email })
```

Comment:
the same about `const { data } = ...`

Reply:
and the same about `users.getPage`

Reply:
And it's very confusing

* I don't have any ideas what `info` will it return. The word `info` is very common and doesn't describe any entity of our system. So I will need to go deep in the implementation to define what I will get.
* When I see `account`, the only thing I think about is the [account entity](https://tokend.gitbook.io/knowledge-base/technical-details/key-entities/accounts), but the method will return me the entity that is not related to it at all

Reply:
Also you use this method to get account id by email. So why don't write the method `getAccountIdByEmail`?

Reply:
And finally, it seems a lot of guys in the team need the same logic (I mean getting accountId by email and vice versa). Let's implement the helpers ASAP so guys can reuse them in their branches

@max-nes

Reply:
Resolved, updated